tests/h/search/index_test.py

```diff
@@ -93,6 +106,129 @@ def test_you_can_filter_annotations_by_creation_time(self, factories, index, sea
         response = search.filter("range", created={"gte": before}).execute()
         assert SearchResponseWithIDs([annotation.id]) == response
 
+    def test_you_can_filter_annotations_by_updated_time(self, factories, index, search):
+        update_time = datetime.datetime.now()
+        annotation = factories.Annotation.build(id="test_annotation_id", updated=update_time)
+
+        index(annotation)
+
+        response = search.filter("range", updated={"gte": update_time}).execute()
+        assert SearchResponseWithIDs([annotation.id]) == response
+
+    def test_you_can_filter_annotations_by_id(self, factories, index, search):
+        annotation = factories.Annotation.build(id="test_ann_id")
+
+        index(annotation)
+
+        response = search.filter("term", id="test_ann_id").execute()
+
+        assert SearchResponseWithIDs([annotation.id]) == response
+
+    @pytest.mark.parametrize('user_search_str', [
+        'acct:someone@example.com',
+        'someone',
+        'someone@example.com'
+    ])
+    def test_you_can_filter_annotations_by_user(self, factories, index, search, user_search_str):
+        annotation = factories.Annotation.build(userid="acct:someone@example.com")
+
+        index(annotation)
+
+        response = search.filter("term", user=user_search_str).execute()
+
+        assert SearchResponseWithIDs([annotation.id]) == response
+
+    def test_you_can_make_aggregations_on_user_raw(self, factories, index, search):
+        annotation = factories.Annotation.build(userid="acct:someone@example.com")
```

Comment:
What are we actually testing here? Does elasticsearch allow creating aggregations on _any_ field, or is there something about the field configuration which is required to support this?

Reply:
This is fine. The purpose of the user_raw (and tags_raw etc) fields is to aggregate by them. We wouldn't want to test that you can filter by them (as we do with the other fields) because that isn't what these fields are for (h only uses them for aggregating).

Hopefully the tests cover the difference between the user analyzer used on the user field, and no analysis being done on the user_raw field, this difference is apparently what necessitates a separate user_raw field for aggregating.

Reply:
+1 to what Sean said ^

Reply:
But @sheetaluk and @seanh, my point is that with this change the test still passes:

```diff
diff --git a/tests/h/search/index_test.py b/tests/h/search/index_test.py
index b04bdaa1a..889c65ffc 100644
--- a/tests/h/search/index_test.py
+++ b/tests/h/search/index_test.py
@@ -143,7 +143,7 @@ class TestIndex(object):
 
         index(annotation)
 
-        a = elasticsearch1_dsl.A('terms', field='user_raw')
+        a = elasticsearch1_dsl.A('terms', field='wibble')
         search.aggs.bucket('user_raw_terms', a)
 
         response = search.execute()
```

On that basis, what exactly are we testing here?